tests/mocha/contextmenu_items_test.js

```diff
@@ -0,0 +1,430 @@
+        block2.initSvg();
+        block2.render();
+        block2.setCollapsed(true);
+
+        expandOption.callback(scope);
+        this.clock.runAll();
+
+        chai.assert.isFalse(block1.isCollapsed(),
+            'Previously expanded block should still be expanded');
+        chai.assert.isFalse(block2.isCollapsed(),
+            'Previously collapsed block should now be expanded');
+      });
+
+      test('Has correct label', function() {
+        chai.assert.equal(expandOption.displayText(), 'Expand Blocks');
+      });
+    });
+
+    suite('Delete', function() {
+      var deleteOption;
+      setup(function() {
+        deleteOption = registry.getItem('workspaceDelete');
+      });
+
+      test('Enabled when blocks to delete', function() {
+        workspace.newBlock('text');
+        chai.assert.equal(deleteOption.preconditionFn(scope), 'enabled');
+      });
+
+      test('Disabled when no blocks to delete', function() {
+        chai.assert.equal(deleteOption.preconditionFn(scope), 'disabled');
+      });
+
+      test('Deletes all blocks after confirming', function() {
+        // Mocks the confirmation dialog and calls the callback with 'true' simulating ok.
+        sinon.stub(Blockly, 'confirm').callsArgWith(1, true);
+
+        workspace.newBlock('text');
+        workspace.newBlock('text');
+        deleteOption.callback(scope);
+        this.clock.runAll();
+        chai.assert.equal(workspace.getTopBlocks(false).length, 0);
+      });
+
+      test('Does not delete blocks if not confirmed', function() {
+        // Mocks the confirmation dialog and calls the callback with 'true' simulating ok.
+        sinon.stub(Blockly, 'confirm').callsArgWith(1, false);
+
+        workspace.newBlock('text');
+        workspace.newBlock('text');
+        deleteOption.callback(scope);
+        this.clock.runAll();
+        chai.assert.equal(workspace.getTopBlocks(false).length, 2);
+      });
+
+      test('No dialog for single block', function() {
+        var confirmStub = sinon.stub(Blockly, 'confirm');
+        workspace.newBlock('text');
+        deleteOption.callback(scope);
+        this.clock.runAll();
+
+        sinon.assert.notCalled(confirmStub);
+        chai.assert.equal(workspace.getTopBlocks(false).length, 0);
+      });
+
+      test('Has correct label for multiple blocks', function() {
+        workspace.newBlock('text');
+        workspace.newBlock('text');
+
+        chai.assert.equal(deleteOption.displayText(scope), 'Delete 2 Blocks');
+      });
+
+      test('Has correct label for single block', function() {
+        workspace.newBlock('text');
+        chai.assert.equal(deleteOption.displayText(scope), 'Delete Block');
+      });
+    });
+  });
+
+  suite('Block Items', function() {
+    var scope;
+    var block;
+
+    setup(function() {
+      block = workspace.newBlock('text');
+      scope = {block: block};
+    });
+
+    teardown(function() {
+      block.dispose();
```

Comment:
I'm not sure if this is necessary because `workspace.dispose` is also called.

Reply:
Removed